### ISSUE TITLE
securityContext is not being respected by the operator

### ISSUE THREAD

**Issue by sudip-moengage** [lyft/flinkk8soperator#174]
Hi,

`FlinkCluster` manifest:
```
apiVersion: flink.k8s.io/v1beta1
kind: FlinkApplication
metadata:
  name: flink-poc
  namespace: flink-operator
  annotations:
  labels:
    environment: development
spec:
  serviceAccountName: flink-poc
  image: <image>
  flinkConfig:
    parallelism.default: 1
    web.upload.dir: /opt/flink
  jobManagerConfig:
    replicas: 1
    resources:
      requests:
        memory: "4Gi"
        cpu: "2"
    nodeSelector:
      eks.moengage.io/namespace: flink-operator
  taskManagerConfig:
    taskSlots: 10
    resources:
      requests:
        memory: "12Gi"
        cpu: "6"
    nodeSelector:
      eks.moengage.io/namespace: flink-operator
  entryClass: "org.apache.flink"
  flinkVersion: "1.9.2"
  jarName: "poc-assembly-0.1-SNAPSHOT.jar"
  parallelism: 1
  entryClass: "abc.demo.ClassName"
  volumes:
    - name: flink-poc
      persistentVolumeClaim:
        claimName: flink-poc
  volumeMounts:
    - name: flink-poc
      mountPath: "/tmp"
  securityContext:
    runAsUser: 9999
    runAsGroup: 9999
    fsGroup: 9999
```

Now, when the pods are getting created:
Task manager pod manifest by executing: `kubectl get pod tm-pod -n flink-operator`
```
apiVersion: v1
kind: Pod
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"flink.k8s.io/v1beta1","kind":"FlinkApplication","metadata":{"annotations":{},"labels":{"environment":"development"},"name":"flink-poc","namespace":"flink-operator"},"spec":{"entryClass":"abc.demo.ClassName","flinkConfig":{"parallelism.default":1,"web.upload.dir":"/opt/flink"},"flinkVersion":"1.9.2","image":"iamge","jarName":"poc-assembly-0.1-SNAPSHOT.jar","jobManagerConfig":{"nodeSelector":{"eks.moengage.io/namespace":"flink-operator"},"replicas":1,"resources":{"requests":{"cpu":"2","memory":"4Gi"}}},"parallelism":1,"securityContext":{"fsGroup":9999,"runAsGroup":9999,"runAsUser":9999},"serviceAccountName":"flink-poc","taskManagerConfig":{"nodeSelector":{"eks.moengage.io/namespace":"flink-operator"},"resources":{"requests":{"cpu":"6","memory":"12Gi"}},"taskSlots":10},"volumeMounts":[{"mountPath":"/tmp","name":"flink-poc"}],"volumes":[{"name":"flink-poc","persistentVolumeClaim":{"claimName":"flink-poc"}}]}}
    kubernetes.io/psp: eks.privileged
  creationTimestamp: "2020-02-20T08:09:52Z"
  generateName: flink-poc-ef5e6a3b-tm-7b57899f65-
  labels:
    environment: development
    flink-app: flink-poc
    flink-app-hash: ef5e6a3b
    flink-deployment-type: taskmanager
    pod-template-hash: 7b57899f65
  name: flink-poc-ef5e6a3b-tm-7b57899f65-9g46w
  namespace: flink-operator
  ownerReferences:
  - apiVersion: apps/v1
    blockOwnerDeletion: true
    controller: true
    kind: ReplicaSet
    name: flink-poc-ef5e6a3b-tm-7b57899f65
    uid: 5f0888aa-53b8-11ea-bf14-0ee952d837f7
  resourceVersion: "32196794"
  selfLink: /api/v1/namespaces/flink-operator/pods/flink-poc-ef5e6a3b-tm-7b57899f65-9g46w
  uid: 5f09c6c5-53b8-11ea-bf14-0ee952d837f7
spec:
  containers:
  - args:
    - taskmanager
    env:
    - name: AWS_METADATA_SERVICE_TIMEOUT
      value: "5"
    - name: AWS_METADATA_SERVICE_NUM_ATTEMPTS
      value: "20"
    - name: APP_NAME
      value: flink-poc
    - name: OPERATOR_FLINK_CONFIG
      value: |
        blob.server.port: 6125
        jobmanager.heap.size: 2097152k
        jobmanager.rpc.port: 6123
        jobmanager.web.port: 8081
        metrics.internal.query-service.port: 50101
        parallelism.default: 1
        query.server.port: 6124
        taskmanager.heap.size: 6291456k
        taskmanager.numberOfTaskSlots: 10
        web.upload.dir: /opt/flink

        jobmanager.rpc.address: flink-poc-ef5e6a3b
        taskmanager.host: $HOST_IP
    - name: FLINK_PROPERTIES
      value: |
        blob.server.port: 6125
        jobmanager.heap.size: 2097152k
        jobmanager.rpc.port: 6123
        jobmanager.web.port: 8081
        metrics.internal.query-service.port: 50101
        parallelism.default: 1
        query.server.port: 6124
        taskmanager.heap.size: 6291456k
        taskmanager.numberOfTaskSlots: 10
        web.upload.dir: /opt/flink

        jobmanager.rpc.address: flink-poc-ef5e6a3b
        taskmanager.host: $HOST_IP
    - name: HOST_NAME
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: metadata.name
    - name: HOST_IP
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: status.podIP
    - name: FLINK_DEPLOYMENT_TYPE
      value: taskmanager
    image: image
    imagePullPolicy: IfNotPresent
    name: taskmanager
    ports:
    - containerPort: 6123
      name: rpc
      protocol: TCP
    - containerPort: 6125
      name: blob
      protocol: TCP
    - containerPort: 6124
      name: query
      protocol: TCP
    - containerPort: 50101
      name: metrics
      protocol: TCP
    resources:
      requests:
        cpu: "6"
        memory: 12Gi
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /tmp
      name: flink-poc
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: flink-poc-token-mfvt5
      readOnly: true
  dnsPolicy: ClusterFirst
  enableServiceLinks: true
  nodeName: ip-100-30-100-30.ec2.internal
  nodeSelector:
    eks.moengage.io/namespace: flink-operator
  priority: 100000000
  priorityClassName: low-priority
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: flink-poc
  serviceAccountName: flink-poc
  terminationGracePeriodSeconds: 30
  tolerations:
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
    tolerationSeconds: 300
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 300
  volumes:
  - name: flink-poc
    persistentVolumeClaim:
      claimName: flink-poc
  - name: flink-poc-token-mfvt5
    secret:
      defaultMode: 420
      secretName: flink-poc-token-mfvt5
```

Same with jobManager. Can you guys point me where I'm doing wrong?

**Comment by sudip-moengage** [comment]
I can see the support for securityContext has been created by @kelly-sm and has already been merged, I'm running v0.4 release of the operator.

**Comment by sudip-moengage** [comment]
I see that is is not release in 0.4, so I'll close this for now.
